code_generator/GdkCodeGenerator/Templates/UnityReaderWriterGenerator.tt

```diff
@@ -23,44 +23,48 @@ namespace <#= qualifiedNamespace #>
         [ComponentId(<#= unityComponentDefinition.Id #>)]
         internal class ReaderWriterCreator : IReaderWriterCreator
         {
-            public IReaderInternal CreateReaderWriter(Entity entity, EntityManager entityManager, ILogDispatcher logDispatcher)
+            public IReaderWriterInternal CreateReaderWriter(Entity entity, EntityManager entityManager, ILogDispatcher logDispatcher)
             {
                 return new ReaderWriterImpl(entity, entityManager, logDispatcher);
             }
         }
 
         [ReaderInterface]
         [ComponentId(<#= unityComponentDefinition.Id #>)]
-        public interface Reader
+        public interface Reader : IReader<<#= componentDetails.TypeName #><# if(fieldDetailsList.Count > 0){ #>, <#= componentDetails.TypeName #>.Update<# } #>>
         {
         }
 
-        public class ReaderWriterImpl : Reader, IReaderInternal
+        [WriterInterface]
+        [ComponentId(<#= unityComponentDefinition.Id #>)]
+        public interface Writer 
+        {
+        }
+
+        internal class ReaderWriterImpl : IReaderWriterInternal, Reader, Writer
         {
             public ReaderWriterImpl(Entity entity, EntityManager entityManager, ILogDispatcher logDispatcher)
             {
             }
+            
+            public Authority Authority => throw new System.NotImplementedException();
 
-            // IReaderInternal methods
-            void IReaderInternal.OnAuthorityChange(Authority authority)
-            {
-                throw new System.NotImplementedException();
-            }
+            public <#= componentDetails.TypeName #> Data => throw new System.NotImplementedException();
 
-            void IReaderInternal.OnComponentUpdate()
-            {
-                throw new System.NotImplementedException();
-            }
-            
-            void IReaderInternal.OnEvent(int eventIndex)
+            public event GameObjectDelegates.AuthorityChanged AuthorityChanged
             {
-                throw new System.NotImplementedException();
+                add => throw new System.NotImplementedException();
+                remove => throw new System.NotImplementedException();
             }
+<# if (fieldDetailsList.Count > 0) #>
+<# { #>
```

Comment:
Put this in the same `<# #>` block as the if - it makes it marginally easier to read imo